app/models/framework/management_charge_calculator/column_based.rb

```diff
@@ -0,0 +1,19 @@
+class Framework
+  module ManagementChargeCalculator
+    class ColumnBased
+      attr_reader :column, :value_to_percentage
+
+      def initialize(column:, value_to_percentage:)
+        @column = column
+        @value_to_percentage = value_to_percentage.with_indifferent_access
+      end
+
+      def calculate_for(entry)
+        column_value_for_entry = entry.data.dig(column).to_s.downcase
```

Comment:
If we are downcasing the column value in the calculator, we'll have to initialise our framework-specific calculators with lowercase keys as well. `with_indifferent_access` is only for symbol - string communication, but still case-sensitive.

```
pry(main)> {:"Lease Rental"=>0.5e0, :"Fleet Management Fee"=>0.5e0, :Damage=>0, :"Other Re-charges"=>0}.with_indifferent_access['lease rental']
=> nil
```

Reply:
Good spot :+1: